Makefile

```diff
@@ -1,28 +1,35 @@
-.PHONY: dist plugin-data backend-plugin-data frontend-plugin-data
-
-dist: plugin-data
+.PHONY: dist 
+dist: #plugin-data
 	rm -rf ./dist
-	cd site && hugo --destination ../dist/html
+	hugo -s site --destination ../dist/html
 
+.PHONY: plugin-data
 plugin-data: backend-plugin-data frontend-plugin-data devtalks-data
 
+.PHONY: backend-plugin-data
 backend-plugin-data:
 	mkdir -p site/data
 	go run ./cmd/plugin-godocs > site/data/PluginGoDocs.json
 	go run ./cmd/plugin-manifest-docs > site/data/PluginManifestDocs.json
 
+.PHONY: frontend-plugin-data
 frontend-plugin-data:
 	rm -rf scripts/mattermost-webapp
 	cd scripts && git clone https://github.com/mattermost/mattermost-webapp.git
 	cd scripts && npm install
 	mkdir -p site/data
 	node scripts/plugin-jsdocs.js > site/data/PluginJSDocs.json
 
+.PHONY: frontend-plugin-data
 devtalks-data:
 	mkdir -p site/data
 ifdef YOUTUBE_API_KEY
 	go run ./cmd/devtalks > site/data/DevTalks.json
 endif
 
-dev:
-	cd site && hugo server -D
+.PHONY: run
+run:
+	hugo server -s site
+
+.PHONY: dev
```

Comment:
Maybe move this out into a `build/legacy.mk` so as to preserve compatibility but clearly deprecate?

Reply:
How would moving it into a different file help with marking it as deprecated?

Reply:
Any reason not to remove this rather than deprecating it?

Also the convention we've been using in other places is `run-server` should we do the same here?

Reply:
though it should be `make run` but 🤷‍♂

Reply:
> Any reason not to remove this rather than deprecating it?

It might disrupt developers in there workflow if a command that they got used to using does not longer exist.

> Also the convention we've been using in other places is `run-server` should we do the same here?

Is this really the case? At last for MM server `make run` is the fire and forget command you want.

Reply:
ahh true, i never use `run`, always use `run-server` for MM. Same with marketplace. Fair enough then `make run` it is.

Regarding `make dev`, at some point whenever we decide to remove `dev`, whoever was used to that workflow will get get disrupted. Either we remove it now or add some sort of warning message that this `make dev` is deprecated and encourage them to use `make run` instead, so atleast they'll know to switch.

Reply:
I would like to add a warning message but didn't found a good way of doing this. Do you have idea?

Reply:
Not much of a `Makefile` expert but after some googling, found this to be working well.
```
dev:
	$(warning **** make dev is DEPRECATED, use make run instead ****)
	make run
```

Reply:
That technically works, but I spawns another process for `make run`. The purist in me doesn't like it, but I will go away sone anyway.

Reply:
> How would moving it into a different file help with marking it as deprecated?

We do this in the server with [`build/legacy.mk`](https://github.com/mattermost/mattermost-server/blob/master/build/legacy.mk). I find it an elegant way to preserve backwards compatibility but avoid cluttering the primary `Makefile`.